doc/fundamentals/lifecycle_of_a_datum.rst

```diff
@@ -0,0 +1,549 @@
+
+Pachyderm’s idea of a “datum” in the context of parallel processing has
+subtle implications for how data is written to output files. It’s
+important to understand under what conditions data will be overwritten
+or merged in output files.
+
+There are four basic rules to how Pachyderm will process your data in
+the pipelines you create.
+
+-  Pachyderm will split your input into individual datums `as you
+   specified <./reference/pipeline_spec.html#the-input-glob-pattern>`__
+   in your pipeline spec
+-  each datum will be processed independently, using `the parallelism
+   you
+   specified <../reference/pipeline_spec.html#parallelism-spec-optional>`__
+   in your pipeline spec, with no guarantee of the order of processing
+-  Pachyderm will merge the output of each pod into each output file,
+   with no guarantee of ordering
+-  the output files will look as if everything was done in one
+   processing step
+
+| If one of your pipelines is written to take all its input, whatever it
+  may be, process it, and put the output into one file, the final output
+  will be one file.
+| The many datums in your pipeline’s input would become one file at the
+  end, with the processing of every datum reflected in that file.
+
+If you write a pipeline to write to multiple files, each of those files
+may contained merged data that looks as if it were all processed in one
+step, even if you would have specified the pipeline to process each
+datum in one container via a ``parallelism_spec`` set to the default
+value of 1.
+
+This sounds a little complicated, but it breaks down simply into three
+distinct relationships between the datums in your pipeline’s input
+repository and in its output repository. We’ll list them, and then go
+into them in detail, pointing to examples that illustrate them.
+
+-  1 to 1: 1 unique datum in to 1 unique datum out
+-  1 to many: 1 unique datum in to many unique datums out
+-  Many to many: many unique datums in map to many datums out.
+
+1:1
+^^^
+
+The best example of this is the `opencv example and beginner’s tutorial <./getting_started/beginner_tutorial.html>`__.
+
+One datum,
+in this case an image,
+is transformed into another datum,
+in this case, another image.
+
+.. image:: 1-1-datum-relationship.gif
+    :align: center
+    :alt: opencv example animation showing 1 unique datum into 1 unique datum out
+
+
+1:many or 1:N
+^^^^^^^^^^^^^
+
+A good example of this is a pipeline designed to split an image into many tiles for further analysis,
+an easy extension of the `opencv example <https://pachyderm.readthedocs.io/en/stable/getting_started/beginner_tutorial.html>`__,
+left as an exercise for you.
+Each tile is unique in the output by necessity;
+you can’t have one image stomping on another image’s tile!
+
+.. image:: 1-N-datum-relationship.gif
+    :align: center
+    :width: 740px
+    :height: 496px
+    :alt: opencv example animation showing 1 unique datum into N unique datums out
+    
+Many:Many or N:M
```

Comment:
The example is good here. I think the gif can be made a bit clearer. Let's tweak in person this week